Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -648,11 +648,22 @@ WebIDL::ExceptionOr<void> fetch_response_handover(JS::Realm& realm, Infrastructu
                 process_body({});
             });
         }
-        // FIXME: 4. Otherwise, fully read response’s body given processBody, processBodyError, and fetchParams’s task
-        //        destination.
+        // 4. Otherwise, fully read response’s body given processBody, processBodyError, and fetchParams’s task
+        //    destination.
         else {
-            // NOTE: This branch is not taken for JS fetch(), which supplies no 'process response consume body' function.
-            (void)process_body_error;
+            // IMPLEMENTATION DEFINED: This is not in the spec, but fully_read will enqueue promise jobs and we need to ensure that there will be an execution context and backup settings object.
+            if (fetch_params.request()->client()) {
+                fetch_params.request()->client()->prepare_to_run_script();
+                fetch_params.request()->client()->prepare_to_run_callback();
+            }
+
+            response.body()->fully_read(move(process_body), move(process_body_error), fetch_params.task_destination());
+
+            // See above.
+            if (fetch_params.request()->client()) {
```

Comment:
```suggestion
            if (auto client = fetch_params.request()->client(); client) {
```
and reuse that below?

Reply:
so like... what I already suggested two days ago?